### DISCUSSION TITLE
How to reduce the code size when use esm?

### DISCUSSION THREAD

**Discussion by realduolaf** [babel/babel#13277] [Q&A]
I want to try my best to reduce the output code size. I'm looking for equivalent code with low code volume, such as not using default export to reduce the use of `_interopRequireDefault`.

Recently I found that es6 grammar will be translated into this:

```js
// source
export const a = 1;

// target
"use strict";

exports.__esModule = true;
exports.a = void 0;
const a = 1;
exports.a = a;
```

Could I make the code directly like this while still using esm syntax?

```js
"use strict";

exports.__esModule = true;
exports.a = 1;
```
Could someone tell me the historical reason for this and the solution? Any help will be appreciated!

**Comment by fedeci** [comment]
> ```js
> 
> "use strict";
> 
> 
> 
> exports.__esModule = true;
> 
> exports.a = 1;
> 
> ```
The above won't let you access `a` from the current file. 

e.g.
```js
exports.__esModule = true;
exports.a = 1;

console.log(a); // would not be possible 
```
You can find more about `void 0` [here](https://stackoverflow.com/questions/54428255/how-do-i-prevent-babel-from-emitting-export-declarations-twice)

**Reply by realduolaf** [reply]
Oh Thanks! I got it. So is there a way to get rid of `exports.a = void 0;`? Maybe I should open a new issue.
